csrc/device_lower/pass/index.cpp

```diff
@@ -1686,6 +1687,122 @@ Val* hardCodedIndexGenerationForStMatrix(
 
   return out;
 }
+#endif
+
+Val* hardCodedIndexGenerationForStMatrix(
+    const LoadStoreOp* ldst,
+    const ForLoop* outer_loop,
+    const int64_t m_tile,
+    const int64_t n_tile,
+    const int64_t m,
+    const int64_t n) {
+  NVF_ERROR(
+      (m_tile == 8 && n_tile == 8) || (m_tile == 16 && n_tile == 8) ||
+          (m_tile == 16 && n_tile == 16),
+      "size not currently supported for stmatrix");
+  Val* out_index = nullptr;
+
+  // A tile_box can be 16x16, 16x8. 8x8
+  // A warp group (128 threads) can work on a number of tile boxes.
+  int64_t max_tile_boxes_in_warp_group_box = 4;
```

Comment:
I was surprised to see warp group mentioned. From the docs (https://docs.nvidia.com/cuda/parallel-thread-execution/#warp-level-matrix-store-instruction-stmatrix) I thought stmatrix only worked at the warp level.

I suspect this might be here because you want to be compatible with the scheduling of the wgmma instruction, but that's just a guess.

Reply:
That's exactly the reason!